### PR TITLE
Module-ify built-in function documentation

### LINKED ISSUE #363: Document built-in modules
The module-ified documentation should update the function documentation to refer to built-in modules, rather than global functions, while also explaining the global function names for users who don't have access to the module system yet.

### PR DESCRIPTION
**Pull request by nex3**
Closes #363

### CODE REVIEW

File: source/documentation/modules.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  ===
+  .button {
+    color: #6b717f;
+    border: 1px solid #878d9a;
+  }
+<% end %>
+
+Sass provides the following built-in modules:
+
+* Any function that Sass doesn't recognize as built-in or user-defined is
```
Comment (jelbourn) [inline source/documentation/modules.html.md.erb]: Move this bullet to the bottom of this list until there's a better place?

File: source/documentation/at-rules/function.html.md.erb [hunk 2]
```diff
@@ -190,20 +190,20 @@ argument is known as an [argument list][].
...
+[`map.keywords()` function]: ../modules/map#keywords
 [map]: ../values/maps
 
 <% fun_fact do %>
-  If you don't ever pass an argument list to the [`keywords()` function][], that
-  argument list won't allow extra keyword arguments. This helps callers of your
-  function make sure they haven't accidentally written any typos in their
+  If you don't ever pass an argument list to the [`map.keywords()` function][],
+  that argument list won't allow extra keyword arguments. This helps callers of
+  your function make sure they haven't accidentally written any typos in their
```
[the remark this answers is not in the document]
Comment (nex3) [reply source/documentation/at-rules/function.html.md.erb]: Done.

File: source/documentation/at-rules/function.html.md.erb [hunk 3]
```diff
@@ -289,7 +293,45 @@ implementations also make it possible to define [custom functions][] in the host
...
-[core library]: ../functions
+[core library]: ../modules
 [custom functions]: ../js-api#functions
-[plain CSS functions]: ../functions/css
+[plain CSS functions]: #plain-css-functions
 [weird syntax]: ../syntax/special-functions
+
+### Plain CSS Functions
+
+Any function call that's not either a user-defined or [built-in](../modules)
```
[the remark this answers is not in the document]
Comment (nex3) [reply source/documentation/at-rules/function.html.md.erb]: I think that's a bit confusing... "evaluates" is being used there to mean "invokes code defined for that function", but we use it elsewhere to mean more generally evaluating an expression, which *does* happen for plain CSS functions. It's just that the result of that evaluation is an unquoted string.

File: source/documentation/at-rules/mixin.html.md.erb [hunk 4]
```diff
@@ -256,24 +256,24 @@ is known as an [argument list][].
...
 
 <% fun_fact do %>
-  If you don't ever pass an argument list to the [`keywords()` function][], that
-  argument list won't allow extra keyword arguments. This helps callers of your
-  mixin make sure they haven't accidentally written any typos in their argument
-  names.
+  If you don't ever pass an argument list to the [`map.keywords()` function][],
+  that argument list won't allow extra keyword arguments. This helps callers of
+  your mixin make sure they haven't accidentally written any typos in their
+  argument names.
```
[the remark this answers is not in the document]
Comment (nex3) [reply source/documentation/at-rules/mixin.html.md.erb]: Done.

File: source/documentation/modules.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+<% content_for :before_introduction do %>
+  <%= partial 'snippets/built-in-module-status' %>
+<% end %>
+
+<% heads_up do %>
+  Before the Sass module system was introduced, all Sass functions were globally
+  available at all times. Many functions still have global aliases (these are
+  listed in their documentation). The Sass team discourages their use and will
+  eventually deprecate them, but for now they're still available for
```
[the remark this answers is not in the document]
Comment (nex3) [reply source/documentation/modules.html.md.erb]: Done.

File: source/documentation/modules.html.md.erb (hunk 1 above)
Comment (nex3) [reply source/documentation/modules.html.md.erb]: Oh, I just meant to delete this when I moved the plain CSS stuff into the `@function` documentation.

**Pull request merged**